### ISSUE TITLE
2.0.0 - Requires space when specifying varchar datatype with length

### ISSUE THREAD

**Issue by fredriv** [sqlfluff/sqlfluff#4537]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff v2.0.0 gives an LT01 error about missing whitespace for a `varchar` data type with specified length, e.g. `varchar(10)`.

This does not result in any error with SQLFluff v1.4.5.

### Expected Behaviour

I expected a data type specification like `varchar(10)` to parse without errors.

### Observed Behaviour

SQLFluff reports an error when parsing the SQL file below:
```
L:   2 | P:  22 | LT01 | Expected single whitespace between data type identifier
                       | and start bracket '('. [layout.spacing]
L:   5 | P:  21 | LT01 | Expected single whitespace between data type identifier
                       | and start bracket '('. [layout.spacing]
```

### How to reproduce

SQL file that fails to parse:

```
create table my_table (
    my_column varchar(10)
);

select null::varchar(10) as my_column
from my_table;
```

### Dialect

Tested with `ansi` and `snowflake`.

### Version

```
> sqlfluff --version
sqlfluff, version 2.0.0

> python --version
Python 3.9.9
```

### Configuration

```
[sqlfluff]
dialect = ansi
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
I _believe_ this is already fixed and will be available in 2.0.1. That release should be out soon, so I'd love if you're able to confirm that when it is.

**Comment by alanmcruickshank** [comment]
Yep - confirmed this is already resolved on `main`. Was sorted by #4528
